src/wasm.h

```diff
@@ -176,6 +176,16 @@ struct Literal {
     }
   }
 
+  int32_t floatPayload() {
+    assert(type == WasmType::f32);
+    return ~0xffc00000u & bit_cast<uint32_t>(f32);
```

Comment:
Add symbolic constants for the mask, and reuse at the other place?

Also, assert the value is a nan?